spring-kafka/src/main/java/org/springframework/kafka/core/DefaultKafkaProducerFactory.java

```diff
@@ -100,13 +108,17 @@ public boolean isRunning() {
 		if (this.producer == null) {
 			synchronized (this) {
 				if (this.producer == null) {
-					this.producer = new CloseSafeProducer<K, V>(new KafkaProducer<K, V>(this.configs));
+					this.producer = new CloseSafeProducer<K, V>(createKafkaProducer());
 				}
 			}
 		}
 		return this.producer;
 	}
 
+	protected KafkaProducer<K, V> createKafkaProducer() {
+		return this.strategy.createKafkaProducer();
```

Comment:
Basically we don't need `this.configs` once we have strategy implementation. Will clean it up and adjust constructors.